### PR TITLE
Code Cleanup

### PR DESCRIPTION
**Pull request by servantftransperfect**
- Remove samples
- Remove useless software

- Remove structure like access to sfmData views, landmarks and intrinsics
- Remove assignment operator to views and intrinsics (which may create new objects without knowledge)

### CODE REVIEW

**Comment by simogasp** [comment]
I would not remove the samples. It's the easiest way for newcomers to test and learn about the different algorithms and is always a good starting point when developing new algorithms.

**Comment by servantftransperfect** [comment]
Samples are a burden to maintain when upgrading the code.

- Who is developing new algorithms similar to the one detailed in the samples ? No new multiview algorithms for 4 years and they were done by insiders.
- There is absolutely no QA on the samples nor test coverage for it.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -78,14 +78,6 @@ jobs:
...
           make test
 
-      - name: Build As Third Party
-        working-directory: ./build_as_3rdparty
-        run: |
-          cmake ../src/samples/aliceVisionAs3rdParty \
-           -DBUILD_SHARED_LIBS:BOOL=ON \
-           -DCMAKE_PREFIX_PATH:PATH="$PWD/../../AV_install;${DEPS_INSTALL_DIR}"
-          make -j$(nproc)
-
```
Comment (fabiencastan) [inline .github/workflows/continuous-integration.yml:88]: We should keep this test to ensure that the cmake is not broken when using it as a third party (like qtAliceVision for instance).

File: src/software/convert/CMakeLists.txt [hunk 2]
```diff
@@ -46,27 +33,3 @@ if(ALICEVISION_BUILD_SFM)
...
-    LINKS aliceVision_system
-          aliceVision_image
-          aliceVision_cmdline
-          Boost::program_options
-          Boost::filesystem
-)
-
-if(ALICEVISION_BUILD_MVS)
-    # Merge two meshes
-    alicevision_add_software(aliceVision_convertMesh
```
Comment (fabiencastan) [inline src/software/convert/CMakeLists.txt:63]: I think this one could be helpful.

**Comment by simogasp** [comment]
> Samples are a burden to maintain when upgrading the code.
> 
> * Who is developing new algorithms similar to the one detailed in the samples ? No new multiview algorithms for 4 years and they were done by insiders.
> * There is absolutely no QA on the samples nor test coverage for it.

Samples are meant as simple test cases for the fundamental bricks of the algorithms. As of today, if you just want e.g. to match two images you have to go through the FeatureMatching software with all the complex interface and data to pass. A simple exe that takes two images and a bunch of parameters is always better than passing through the whole sfmdata etc.
And say that one wants to develop a new matching algorithm, he can use the sample as a base (or create a new) one to test it on a simple case, without the hustle and pain of passing through the software and/or meshroom.

Because nobody has contributed yet is not a good reason to remove it, if anything it won't make it more likely. If there are samples where they can start from without passing through the complexity of software exes, it certainly won't discourage them.

**Pull request closed without merging**